pinot-controller/src/main/java/org/apache/pinot/controller/validation/RealtimeSegmentValidationManager.java

```diff
@@ -117,8 +117,18 @@ protected void processTable(String tableNameWithType, Context context) {
 
   private void runSegmentLevelValidation(TableConfig tableConfig, PartitionLevelStreamConfig streamConfig) {
     String realtimeTableName = tableConfig.getTableName();
+
     List<SegmentZKMetadata> segmentsZKMetadata = _pinotHelixResourceManager.getSegmentsZKMetadata(realtimeTableName);
 
+    // Delete tmp segments
+    try {
+      long numDeleteTmpSegments = _llcRealtimeSegmentManager.deleteTmpSegments(realtimeTableName, segmentsZKMetadata);
```

Comment:
nit: one alternative for this may be to check `isTmpSegmentAsyncDeletionEnabled` outside `deleteTmpSegments` and avoid calling `deleteTmpSegments` altogether if the feature is not enabled. Also `deleteTmpSegments` could add a `Precondition(isTmpSegmentAsyncDeletionEnabled(), "called even though not enabled")`. Though this also works so feel free to ignore.